trestle/core/rule_parameters_validator.py

```diff
@@ -0,0 +1,125 @@
+# -*- mode:python; coding:utf-8 -*-
+
+# Copyright (c) 2023 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Validate by confirming rule parameter values are consistent."""
+import logging
+import pathlib
+from typing import List, Optional
+
+from trestle.common.common_types import TopLevelOscalModel
+from trestle.common.common_types import TypeWithSetParams
+from trestle.common.list_utils import as_list, deep_set, get_item_from_list
+from trestle.core.catalog.catalog_interface import CatalogInterface
+from trestle.core.profile_resolver import ProfileResolver
+from trestle.core.validator import Validator
+from trestle.oscal.ssp import ByComponent, ImplementedRequirement, Statement, SystemComponent, SystemSecurityPlan
+
+logger = logging.getLogger(__name__)
+
+
+class RuleParametersValidator(Validator):
+    """Validator to confirm all rule parameter values are consistent."""
+
+    def __init__(self):
+        """Initialize rule param values dictionary."""
+        self._rule_param_values_dict = {}
+
+    def _iter_and_add(
+        self,
+        by_components: List[ByComponent],
+        components: List[SystemComponent],
+        cat_int: CatalogInterface,
+        imp_requirement: ImplementedRequirement,
+        statement: Optional[Statement] = None
+    ) -> None:
+        """
+        Iterate all by components in an object and add the rule shared parameter values to list.
+
+        args:
+            by_components: A list of by components.
+            components: Components defined at System implementation level.
+            cat_int: Instance of catalog interface with controls catalog loaded.
```

Comment:
You don't really need the components list because you can build the dict just using the comp_uuids.  So this can be simplified.  The comp title is useful for a debug or warning msg but not essential

Reply:
done